### DISCUSSION TITLE
Mögliche Ursachen für http-status 401 (unauthorized) bei Senden der signierten Challenge und öffentlichem Zertifikat??

### DISCUSSION THREAD

**Discussion by mge-medisoftware** [Digitaler-Impfnachweis/certification-apis#144] [Q&A] [answered]
Diese Woche wurden von unseren Praxen ca. 23.000 Impfnachweise erfolgreich erstellt. Seltene, typische behebbare Problem sind:
- routing in TI funktioniert nicht
- certificate revocation nicht möglich (kein Internet, firewall..)

Wir kennen sehr genau die Fehlerhäufigkeiten bedingt durch Einsatz von graylog.

Beim Holen des JWT-Tokens bzw. beim Senden der signierten Challenge und des öffentlichem Zertifikates
gibt es bei genau drei Praxen  den Http-Status 401 (unauthorized).
Das bekommen wir nicht behoben. Was sind mögliche Ursachen??

**Comment by HerrTete** [comment]
@mge-medisoftware Haben Sie eine Lösung gefunden? Wir können das gleiche Verhalten reproduzieren.

**Reply by HerrTete** [reply]
Wir konnten das Problem umgehen indem wir für jede Anfrage die Verbindung neu öffnen und schließen. Sobald die TLS-Session für mehrere SMC-Bs verwendet wird bekommen wir beim Abholen des States eine 401.
@oliver-steinbrecher Ist das so gedacht oder kommt da noch ein fix im Backend?

**Comment by oliver-steinbrecher** [comment]
Welche ProfessionOids liegen im SMCB-Certificate vor ?

**Reply by mge-medisoftware** [reply]
Da ist z.B. bei einem Urologen tatsächlich die falsche professionOID eines Psychotherapeuten auf der SMC-B....
ProfessionInfo: Betriebsstätte Psychotherapeut
Profession OID: 1.2.276.0.76.4.52
subject-name auf der SMC-B: CN=Urologie im Zentrum

**Comment by oliver-steinbrecher** [comment, marked as answer]
Bisher ist nur die OID `1.2.276.0.76.4.50` freigeschaltet - siehe auch https://github.com/Digitaler-Impfnachweis/certification-apis/blob/master/SMCB-Authentication.md#3-perform-signature-operation-with-smcb . Weitere OIDs können mit aufgenommen werden, müssen aber explizit von den Stakeholdern freigeben und konfigurativ freigeschaltet werden

**Reply by richSpiegel** [reply]
Betrifft das auch die RU? In dem Link von Ihnen steht ja explizit drin: "Test environments are not limited in respect to the ProfessionOID."

**Reply by ManuelB** [reply]
@richSpiegel I was able in RU to create certificates with a hospital SMC-B.

**Reply by richSpiegel** [reply]
Still won´t work with KOPS and hospital SMC-B! Are you testing with a real card and RU-connector?

**Reply by richSpiegel** [reply]
Gibt es irgendeinen Support für das Problem 401-Problem von Ubirch? Hänge noch immer in der Challenge für KOPS SMC-B mit Krankenhauskarte! Keine Idee, was das Problem sein soll!
Bekomme folgende Location:
https://id.certify.demo.ubirch.com/auth/realms/bmg-ti-certify/login-actions/authenticate?session_code=P7jpKmx6oSvj4p5qy-YpQB9VDWM41AiTibJBimJkK2o&execution=a4825c72-879a-4f96-8ed5-d1c1b0e6bf70&client_id=user-access-ti-simulator&tab_id=8oFlxOk1wbE&auth_session_id=a41c9293-ad92-4470-82d5-79a1f194f1e5

**Reply by oliverschaefer** [reply]
@oliver-steinbrecher Mit dem heutigen Ministerpräsidenten-Beschluss sollte die OID 1.2.276.0.76.4.51 für Zahnarztpraxen im Backend aufgenommen werden. Siehe auch #205

**Reply by oliver-steinbrecher** [reply]
Krankenhäuser befinden sich gerade in der frühen Integrationsphase. Zahnärzte werden wir auch betrachten. Durch die heterogenen Umgebungen sind aber weitere Sicherheitsmaßnahmen auf deren notwendig zu prüfen.

**Reply by oliverschaefer** [reply]
@oliver-steinbrecher Bei den Zahnärzten ist die TI Anbindung von allen Berufsgruppen am weitesten. Diese sind damit weit weniger heterogen aufgestellt als Klinkumgebungen

**Comment by richSpiegel** [comment]
Thanks for your help. Will check it out!

Von: Manuel Blechschmidt ***@***.***
Gesendet: Friday, 6 August 2021 10:58
An: Digitaler-Impfnachweis/certification-apis ***@***.***>
Cc: richSpiegel ***@***.***>; Mention ***@***.***>
Betreff: Re: [Digitaler-Impfnachweis/certification-apis] Mögliche Ursachen für http-status 401 (unauthorized) bei Senden der signierten Challenge und öffentlichem Zertifikat?? (#144)


@richSpiegel<https://github.com/richSpiegel> I was able to create certificates with a hospital SMC-B.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Digitaler-Impfnachweis/certification-apis/discussions/144#discussioncomment-1138306>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALUPGBMMOB532ZROCJ3A4MLT3OPY5ANCNFSM47J257PQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.

**Reply by ManuelB** [reply]
Yes, I use a real card and a real RU connector.

You can see in the followuing video waht I am doing: https://www.youtube.com/watch?v=C0KfzxP-HzU
